templates/rust/src/client.rs.twig

```diff
@@ -0,0 +1,254 @@
+                    );
+                  }
+                }
+
+                headers
+            }
+            None => self.headers.clone(),
+        };
+
+        // Now start building request with reqwest
+        let method_type = match method {
+            "GET" => reqwest::Method::GET,
+            "POST" => reqwest::Method::POST,
+            "OPTIONS" => reqwest::Method::OPTIONS,
+            "PUT" => reqwest::Method::PUT,
+            "DELETE" => reqwest::Method::DELETE,
+            "HEAD" => reqwest::Method::HEAD,
+            "PATCH" => reqwest::Method::PATCH,
+            _ => reqwest::Method::GET,
+        };
+
+        let mut request = self
+            .client
+            .request(method_type, self.endpoint.join(&format!("{}{}", "v1", path)).unwrap());
+
+        match params {
+            Some(data) => {
+                let flattened_data = flatten(FlattenType::Normal(data.clone()), None);
+
+                // First flatten the data and feed it into a FormData
+                if content_type.starts_with("multipart/form-data") {
+                    let mut form = reqwest::blocking::multipart::Form::new();
+
+                    for (k, v) in flattened_data.clone() {
+                        match v {
+                            ParamType::Bool(data) => {
+                                form = form.text(k, data.to_string());
+                            }
+                            ParamType::String(data) => form = form.text(k, data),
+                            ParamType::FilePath(data) => form = form.file(k, data).unwrap(),
+                            ParamType::Number(data) => form = form.text(k, data.to_string()),
+                            // This shouldn't be possible due to the flatten function, so we won't handle this for now
+                            ParamType::Array(_data) => {
+                                //todo: Feed this back into a flatten function if needed
+                            },
+                            ParamType::Object(_data) => {
+                              // Same for this
+                            }
+                        }
+                    }
+                    request = request.multipart(form);
+                }
+
+                if content_type.starts_with("application/json") && method != "GET" {
+                    request = request.json(&data);
+                }
+
+                if method == "GET" {
+                    request = request.query(&queryize_data(flatten(FlattenType::Normal(data), None)));
+                }
+            }
+            None => {}
+        }
+
+        request = request.headers(request_headers);
+
+        match request.send() {
+            Ok(data) => {
+                if data.status().is_success() {
+                    Ok(data)    
+                } else {
+                    // Format error
+                    Err(match data.json() {
+                        Ok(data) => data,
```

Comment:
Once error has occurred should the `Ok(data)` be called? or just throwing exception?

Reply:
The Ok(data) is for the match phrase called on data.json() since it returns a Option. We have to handle if it's a Ok or Err in order to get to the data